### PR TITLE
fix(Core/Query): Update GroupMgr query to support MySQL 8.0

### PR DESCRIPTION
**Pull request by iThorgrim-Hub**
DELETE GROUP < 1 Members

## CHANGES PROPOSED:
-  [Update Query viewed in TrinityCore](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Groups/GroupMgr.cpp#L136)

## TESTS PERFORMED:
- Builds
- Starting Server With MySQL 8.0 & MySQL 5.6

## HOW TO TEST THE CHANGES:
- Just start ur server

## Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Kitzunu** [comment]
Doing my work for me. Thank you for the PR :D

File: src/server/game/Groups/GroupMgr.cpp [hunk 1]
```diff
@@ -92,11 +92,11 @@ void GroupMgr::LoadGroups()
 
         // Delete all groups whose leader does not exist
         CharacterDatabase.DirectExecute("DELETE FROM `groups` WHERE leaderGuid NOT IN (SELECT guid FROM characters)");
+
         // Delete all groups with less than 2 members (or less than 1 for lfg groups)
-        // TODO: This query is not allowed in MySQL 8.0
-        CharacterDatabase.DirectExecute("DELETE `groups` FROM `groups` LEFT JOIN ((SELECT guid, count(*) as cnt FROM group_member GROUP BY guid) t) ON groups.guid = t.guid WHERE t.guid IS NULL OR (t.cnt<=1 AND groups.groupType <> 12)");
+        CharacterDatabase.DirectExecute("DELETE FROM `groups` WHERE guid NOT IN (SELECT guid FROM group_member GROUP BY guid HAVING COUNT(guid) > 1)");
```
Comment (Kitzunu) [inline src/server/game/Groups/GroupMgr.cpp:97]: Are you sure this should be > 1 (greater than one) and not <= 1 (Less or equal to one)?

**Comment by FrancescoBorzi** [comment]
> HOW TO TEST THE CHANGES:
> Just start ur server

eemh no, we need to test when actually the query is called:

> // Delete all groups with less than 2 members (or less than 1 for lfg groups)

and make sure that everything is ok

**Comment by iThorgrim-Hub** [comment]
> > HOW TO TEST THE CHANGES:
> > Just start ur server
> 
> eemh no, we need to test when actually the query is called:

When the server turns on.

File: src/server/game/Groups/GroupMgr.cpp [hunk 2]
```diff
@@ -94,7 +94,7 @@ void GroupMgr::LoadGroups()
         CharacterDatabase.DirectExecute("DELETE FROM `groups` WHERE leaderGuid NOT IN (SELECT guid FROM characters)");
 
         // Delete all groups with less than 2 members (or less than 1 for lfg groups)
-        CharacterDatabase.DirectExecute("DELETE FROM `groups` WHERE guid NOT IN (SELECT guid FROM group_member GROUP BY guid HAVING COUNT(guid) > 1)");
+        CharacterDatabase.DirectExecute("DELETE FROM `groups` WHERE guid NOT IN (SELECT guid FROM group_member GROUP BY guid HAVING COUNT(guid) >= 1)");
```
Comment (FrancescoBorzi) [inline src/server/game/Groups/GroupMgr.cpp:97]: @iThorgrim-Hub @Kitzunu why this change?

Comment (FrancescoBorzi) [reply src/server/game/Groups/GroupMgr.cpp:97]: also where is this condition applied `(or less than 1 for lfg groups)` ?

Comment (FrancescoBorzi) [reply src/server/game/Groups/GroupMgr.cpp:97]: PS: in TC it's `HAVING COUNT(guid) > 1)` why should it be different for us?

**Comment by FrancescoBorzi** [comment]
tested on Ubuntu 20.04 with MySQL 8. The server starts normally.

File: src/server/game/Groups/GroupMgr.cpp (hunk 2 above)
Comment (FrancescoBorzi) [reply src/server/game/Groups/GroupMgr.cpp]: solved in chat
